### ISSUE TITLE
Wandblogger not logging train loss after every step

### ISSUE THREAD

**Issue by Gladiator07** [PyTorchLightning/pytorch-lightning#8559]
## 🐛 Bug

I am using wandb with Pytorch Lightning. I am logging train/loss, val/loss, train/metric, val/metric. Everything is logged properly to wandb dashboard except the train/loss (after every step).

Here's the main lightning module:

`class ImageClassification(pl.LightningModule):
    def __init__(self, model):
        super().__init__()
        self.model = model
        self.criterion = nn.BCEWithLogitsLoss()
        self.lr = CFG['lr']
    
    def forward(self, x):
        output = self.model(x)
        return output
    
    def configure_optimizers(self):
        self.optimizer = torch.optim.AdamW(self.model.parameters(), lr=CFG['lr'])
        return self.optimizer
    
    def training_step(self, batch, batch_idx):
        images, targets = batch['image'], batch['target']
        output = self.model(images)
        loss = self.criterion(output.view(-1), targets)
        
        self.log('train/loss', loss, logger=True)  # the thing that is not being logged

        try:
            auc = roc_auc_score(targets.detach().cpu(), output.sigmoid().detach().cpu())
            self.log("train/auc", auc, prog_bar=True, logger=True)
        except:
            pass
        
        return {
            "loss": loss,
            "predictions": output,
            "targets" : targets
        }
    
    def training_epoch_end(self, outputs):

        preds = []
        targets = []

        for output in outputs:
            preds += output['predictions']
            targets += output['targets']
        
        targets = torch.stack(targets)
        preds = torch.stack(preds)

        train_auc = roc_auc_score(targets.detach().cpu(), preds.sigmoid().detach().cpu())
        self.log("train/auc_epoch", train_auc,logger=True)
    
    def validation_step(self, batch, batch_idx):
        images, targets = batch['image'], batch['target']
        output = self.model(images)
        loss = self.criterion(output.view(-1), targets)

        self.log('val/loss', loss,prog_bar=True, logger=True)

        return {
            "predictions": output,
            "targets": targets
        }
    
    def validation_epoch_end(self, outputs):

        preds = []
        targets = []

        for output in outputs:
            preds += output['predictions']
            targets += output['targets']
        
        targets = torch.stack(targets)
        preds = torch.stack(preds)
    
        val_auc = roc_auc_score(targets.detach().cpu(), preds.sigmoid().detach().cpu())
        self.log("val/auc_epoch", val_auc,prog_bar=True,logger=True)
    
    def test_step(self, batch, batch_idx):
        images = batch['image']
        output = self.model(images)
        return output`

**Comment by awaelchli** [comment]
is the logging interval small enough? I.e., `Trainer(log_every_n_step=n)` where n must be smaller than `len(dataloader)`.

**Comment by awaelchli** [comment]
```python
try:
        auc = roc_auc_score(targets.detach().cpu(), output.sigmoid().detach().cpu())
        self.log("train/auc", auc, prog_bar=True, logger=True)
    except:
        pass
```

If an exception raises here, it will be silently ignored and nothing gets logged. Please try to remove the try-except block here and see if it raises an exception.

**Comment by Gladiator07** [comment]
Thanks, changing n to 1 in `Trainer(log_every_n_step=n)` solved my problem. I was testing my code with smaller data so it couldn't log as you said. I am closing this issue.

**Comment by awaelchli** [comment]
Happy to help. Glad you found the issue. Lightning should also print a warning in that case, you may have missed it or you are not on the latest version. 
cheers
